src/sentry/api/endpoints/group_events.py

```diff
@@ -38,63 +43,78 @@ def get(self, request, group):
+        from sentry.api.serializers.models.event import SnubaEvent
+        from sentry.utils.snuba import raw_query
 
         try:
-            environment = self._get_environment_from_request(
-                request,
-                group.project.organization_id,
-            )
-        except Environment.DoesNotExist:
-            return respond(events.none())
-
-        raw_query = request.GET.get('query')
-
-        if raw_query:
-            try:
-                query_kwargs = parse_query([group.project], raw_query, request.user)
-            except InvalidQuery as exc:
-                return Response({'detail': six.text_type(exc)}, status=400)
+            environment = self._get_environment(request, group)
+            query, tags = self._get_search_query_and_tags(request, group, environment)
+        except InvalidQuery as exc:
+            return Response({'detail': six.text_type(exc)}, status=400)
+        except NoResults:
+            return Response([])
+
+        conditions = []
+        if query:
+            message_condition = [['positionCaseInsensitive', ['message', "'%s'" % (query,)]], '!=', 0]
+            if is_event_id(query):
+                or_condition = [message_condition, ['event_id', '=', query]]
+                conditions.append(or_condition)
             else:
-                query = query_kwargs.pop('query', None)
-                tags = query_kwargs.pop('tags', {})
-        else:
-            query = None
-            tags = {}
+                conditions.append(message_condition)
 
-        if environment is not None:
-            if 'environment' in tags and tags['environment'] != environment.name:
-                # An event can only be associated with a single
-                # environment, so if the environment associated with
-                # the request is different than the environment
-                # provided as a tag lookup, the query cannot contain
-                # any valid results.
-                return respond(events.none())
-            else:
-                tags['environment'] = environment.name
+        if tags:
+            conditions.extend([[u'tags[{}]'.format(k), '=', v] for (k, v) in tags.items()])
+
+        now = timezone.now()
+        data_fn = partial(
+            # extract 'data' from raw_query result
+            lambda *args, **kwargs: raw_query(*args, **kwargs)['data'],
+            start=now - timedelta(days=90),
+            end=now,
+            conditions=conditions,
+            filter_keys={
+                'project_id': [group.project_id],
+                'issue': [group.id]
+            },
+            selected_columns=SnubaEvent.selected_columns,
+            orderby='-timestamp',
+            referrer='api.group-events',
+        )
+
+        return self.paginate(
+            request=request,
+            on_results=lambda results: serialize(
+                [SnubaEvent(row) for row in results], request.user),
```

Comment:
I added a change to make `SnubaEvent` fetch all tags. This PR is supposed to provide a compatible implementation for the group events page so I should do it here. 

It will all be replaced by the aformentioned PR, but who knows when that will get reviewed/shipped